app/src/main/java/ch/blinkenlights/android/vanilla/SlidingPlaybackActivity.java

```diff
@@ -227,15 +225,28 @@ protected QueryTask buildQueryFromIntent(Intent intent, boolean empty, LibraryAd
 		else
 			projection = empty ? Song.EMPTY_PROJECTION : Song.FILLED_PROJECTION;
 
-		long id = intent.getLongExtra("id", LibraryAdapter.INVALID_ID);
 		QueryTask query;
-		if (allSource != null) {
-			query = allSource.buildSongQuery(projection);
-			query.data = id;
-		} else if (type == MediaUtils.TYPE_FILE) {
-			query = MediaUtils.buildFileQuery(intent.getStringExtra(LibraryAdapter.DATA_FILE), projection, true /* recursive */);
+		Object id = intent.getExtras().get("id");
```

Comment:
We should either:

* make 'id' to always be an array
* ...or use a new key for cases where we pass in an array (get("id_list") ? )

Reply:
Ok I went with the second approach since there are too many places using "id". See the update.